arwen/elrondapi/managedei.go

```diff
@@ -0,0 +1,1035 @@
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedUpgradeContract", v1_4_managedUpgradeContract, C.v1_4_managedUpgradeContract)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedUpgradeFromSourceContract", v1_4_managedUpgradeFromSourceContract, C.v1_4_managedUpgradeFromSourceContract)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedAsyncCall", v1_4_managedAsyncCall, C.v1_4_managedAsyncCall)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedESDTTransfers", v1_4_managedESDTTransfers, C.v1_4_managedESDTTransfers)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetESDTBalance", v1_4_managedGetESDTBalance, C.v1_4_managedGetESDTBalance)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetESDTTokenData", v1_4_managedGetESDTTokenData, C.v1_4_managedGetESDTTokenData)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetReturnData", v1_4_managedGetReturnData, C.v1_4_managedGetReturnData)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetPrevBlockRandomSeed", v1_4_managedGetPrevBlockRandomSeed, C.v1_4_managedGetPrevBlockRandomSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetBlockRandomSeed", v1_4_managedGetBlockRandomSeed, C.v1_4_managedGetBlockRandomSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetStateRootHash", v1_4_managedGetStateRootHash, C.v1_4_managedGetStateRootHash)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("managedGetOriginalTxHash", v1_4_managedGetOriginalTxHash, C.v1_4_managedGetOriginalTxHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return imports, nil
+}
+
+//export v1_4_managedSCAddress
+func v1_4_managedSCAddress(context unsafe.Pointer, destinationHandle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetSCAddress
+	metering.UseGas(gasToUse)
+
+	scAddress := runtime.GetSCAddress()
+
+	managedType.SetBytes(destinationHandle, scAddress)
+}
+
+//export v1_4_managedOwnerAddress
+func v1_4_managedOwnerAddress(context unsafe.Pointer, destinationHandle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	blockchain := arwen.GetBlockchainContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetOwnerAddress
+	metering.UseGas(gasToUse)
+
+	owner, err := blockchain.GetOwnerAddress()
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	managedType.SetBytes(destinationHandle, owner)
```

Comment:
Might be safe without copying the bytes, but it might also not be safe. I suggest copying anyway.